analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UseReturnStatement.vb

```diff
@@ -17,37 +19,44 @@
+
     Public Function CaseInsensitive() As Integer
         CASEINSENSITIVE = 42  ' Noncompliant
     End Function
 
     Public Shared Function SharedFunction() As Integer
         SharedFunction = 42  ' Noncompliant
     End Function
-    
+
     Public Function ExplictlyReturnDefaultReturnValue() As Integer
         Return ExplictlyReturnDefaultReturnValue ' Noncompliant {{Do not make use of the implicit return value.}}
     End Function
-        
+
     Public Function ReadAssignementStatements() As Integer
-        dim value as Integer = ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
-        value += ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
-        value -= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
-        value *= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
-        value /= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
-        value \= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
-        value &= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
-        value ^= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
+        Dim value As Integer = ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
+        value += ReadAssignementStatements  ' Noncompliant {{Do not make use of the implicit return value.}}
+        value -= ReadAssignementStatements  ' Noncompliant {{Do not make use of the implicit return value.}}
+        value *= ReadAssignementStatements  ' Noncompliant {{Do not make use of the implicit return value.}}
+        value /= ReadAssignementStatements  ' Noncompliant {{Do not make use of the implicit return value.}}
+        value \= ReadAssignementStatements  ' Noncompliant {{Do not make use of the implicit return value.}}
+        value &= ReadAssignementStatements  ' Noncompliant {{Do not make use of the implicit return value.}}
+        value ^= ReadAssignementStatements  ' Noncompliant {{Do not make use of the implicit return value.}}
         value <<= ReadAssignementStatements ' Noncompliant {{Do not make use of the implicit return value.}}
         value >>= ReadAssignementStatements ' Noncompliant {{Do not make use of the implicit return value.}}
         Return value
     End Function
-    
-     Public Function ImplementedInterfaceMethod() As Integer Implements IInterface.ImplementedInterfaceMethod
+
+    Public Function ImplementedInterfaceMethod() As Integer Implements IInterface.ImplementedInterfaceMethod
         ImplementedInterfaceMethod = 42 ' Noncompliant
     End Function
 
+    Private Function ArgumentName() As Integer
+        WithExplicitArgumentName(ArgumentName:=ArgumentName)    ' Noncompliant
```

Comment:
why is this Noncompliant? isn't this a named argument assignment rather than assigning a value to the function name?

Reply:
First `ArgumentName` is really argument name of the `WithExplicitArgumentName(ArgumentName As Integer)` method.
`Name:=Value` is equivalent to `name : value` explicit parameter naming when you need to pass them out of order.
Second `ArgumentName` is the noncompliant stuff. It's the hidden variable declared by enclosing function name.

Reply:
I made this more explicit by renaming the parameter.